### ISSUE TITLE
Reloading image after getting the value

### ISSUE THREAD

**Issue by vrtulka23** [blackary/streamlit-image-coordinates#19]
I use the clicked coordinates to display points on the image.
However, currently the image is being displayed before I get the "value" and I have to reload the script in order to show changes on the image.
This leads to a very annoying flickering after each click.
It would be nice to have an option to update the image immediately after acquiring the coordinate values.

**Comment by vrtulka23** [comment]
Just to clarify. Ideally one does not have to use "st.rerun()" to update the image.
Cannot you use e.g. st.empty() to clear and refresh the image?

[st.empty()](https://docs.streamlit.io/develop/api-reference/layout/st.empty)

**Comment by vrtulka23** [comment]
I have an idea how to solve this.
You could update image URL in this method by simply adding "?salt=<random_number>" into the following method:
https://github.com/blackary/streamlit-image-coordinates/blob/1ecfe31d4fd6e05508f2673c810cdaa779ae23e1/src/streamlit_image_coordinates/frontend/main.js#L16

This would force the browser to reload the image every time user click on it and load the newest version.

**Comment by Shade5** [comment]
+1 on this. Using `st.rerun` works as mentioned [here](https://github.com/blackary/streamlit-image-coordinates/issues/18), but it's very annoying to use as the screen flickers on very click

**Comment by AliKhoda** [comment]
@blackary, Is it possible to add an on_click callback parameter to streamlit_image_coordinates()?

**Comment by blackary** [comment]
Yes, if we're willing to limit to streamlit>=1.36.0, custom components can have an on_change, which seems like a good idea to me. It would make this problem much simpler to solve.

**Comment by blackary** [comment]
The `on_click` callback added in v0.4.0 addresses this. Instead of polling `value` and re-running to update the image, pass a callback directly:

```python
def update_image():
    value = st.session_state["img"]  # the key you set
    # mutate your image based on value, store in session_state

streamlit_image_coordinates(image, key="img", on_click=update_image)
```

The callback fires immediately when a click is registered, before the next full script re-run, so there's no need to reload or wait. Closing since the feature is now available.
